akka-http-core/src/main/resources/reference.conf

```diff
@@ -87,6 +87,11 @@ akka.http {
     # error details.
     verbose-error-messages = off
 
+    # Puts the server into a reverse-proxy mode
+    # This, effectively, disables special handling of the Expect: 100-continue
+    # behaviour
+    proxy-mode = off
+
```

Comment:
I expect that this will be a fairly contentious issue. The problem is that a "correct" fix (e.g. a new directive) would have a big challenge ahead of it. The places where the change needs to be made do not have all of the contextual information required in order to make that change.  Hence, the configuration setting.